apps/demos/Demos/Chat/Customization/Angular/app/app.component.ts

```diff
@@ -0,0 +1,61 @@
+import { NgModule, Component, enableProdMode } from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+import { platformBrowserDynamic } from '@angular/platform-browser-dynamic';
+
+import { DxChatModule, DxCheckBoxModule, DxSelectBoxModule } from 'devextreme-angular';
+import { User, Message, MessageEnteredEvent } from 'devextreme/ui/chat';
+import { AppService } from './app.service';
+import { Observable } from 'rxjs';
+
+if (!/localhost/.test(document.location.host)) {
+  enableProdMode();
+}
+
+let modulePrefix = '';
+// @ts-ignore
+if (window && window.config.packageConfigPaths) {
+  modulePrefix = '/app';
+}
+
+@Component({
+  selector: 'demo-app',
+  templateUrl: `.${modulePrefix}/app.component.html`,
+  styleUrls: [`.${modulePrefix}/app.component.css`],
+})
+export class AppComponent {
+  currentUser: User;
+  supportAgent: User;
+  messages$: Observable<Message[]>;
+  showAvatar=true;
+  showUserName=true;
+  showDayHeaders=true;
+  showMessageTimestamp=true;
```

Comment:
Okay, now I see u really can omit this state properties while u can bind checkboxes/selectboxes values directly to a chat property (how u did for disabled). But please test it works well on az-server

Reply:
[37c8016](https://github.com/DevExpress/DevExtreme/pull/28482/commits/37c8016387efd0b5991f5f9cb043fba1c686bd51)